integr8ly/gitea-operator
# Add e2e tests

The following changes includes:
- Addition of e2e tests. There are 2 tests included, one for deploying gitea with oauthproxy and another without.
- Move Gitea RBAC files to individual files: `service_account`, `roles` and `role_binding`. This is required by the operator-sdk framework for e2e testing for v0.0.7+
- Move the crd definition to `deploy/crd/` folder. This is required by the operator-sdk framework for e2e testing for v0.0.7+
- Addition of rule for running e2e tests locally in the Makefile
- Update of operator-sdk version from 0.0.7 to 0.1.1
- Vendor changes

The e2e tests are based on the [memcached-operator](https://github.com/operator-framework/operator-sdk-samples/tree/master/memcached-operator) example and operator-sdk e2e testing [documentation](https://github.com/operator-framework/operator-sdk/blob/master/doc/test-framework/writing-e2e-tests.md).

**Verification**
1. Login to Openshift
2. Run `make test-e2e-local`
3. Ensure that all tests are passing
